src/ORM/FieldType/DBForeignKey.php

```diff
@@ -74,8 +81,21 @@ public function scaffoldFormField($title = null, $params = null)
         // Don't scaffold a dropdown for large tables, as making the list concrete
         // might exceed the available PHP memory in creating too many DataObject instances
         $threshold = self::config()->get('dropdown_field_threshold');
-        if ($list->count() < $threshold) {
-            $field = new DropdownField($this->name, $title, $list->map('ID', $titleField));
+
+        // Add the count of the list to a cache for subsequent calls
+        if (!isset(static::$foreignListCache[$hasOneClass])) {
```

Comment:
How would this work between model calls? Would you get cache bleed between them?

Reply:
The calls are literally `DataList::create($hasOneClass)->count()`. I don't think it matters where it comes from, the result should be the same regardless AFAICS.

I think the only possible avenue for trouble comes if you have something that adds objects after the first call...

Reply:
Right, but what about HasManyLists for example, e.g. `$blog->Categories()->count()` and `$blogPost->Categories()->count()`. I guess that kind of logic isn't affected by this change since it would be internally filtered, but just spitballing to check

Reply:
Yeah this is only for "hasOne"s. Line 65 to 68 fetch the hasOne relation with `$schema->hasOneComponent` and returning if it doesn't exist.

Reply:
Cool. Tests are green too which would've caught something if my concern was valid. LGTM.